jetty-start/src/main/java/org/eclipse/jetty/start/fileinits/MavenLocalRepoFileInitializer.java

```diff
@@ -219,6 +290,65 @@ public Coordinates getCoordinates(URI uri)
         return coords;
     }
 
+    protected void download(Coordinates coords, Path destination)
+        throws IOException
+    {
+        if (coords.version.endsWith("-SNAPSHOT"))
+        {
+            Path localRepoMetadataPath = localRepositoryDir.resolve(coords.toMetadataPath());
+            if (isMetadataStale(localRepoMetadataPath))
+            {
+                // Grab a fresh copy of the metadata xml
+                URI mavenMetadataURI = coords.toSnapshotMetadataXmlURI();
+                super.download(mavenMetadataURI, localRepoMetadataPath);
+            }
+
+            if (Files.exists(localRepoMetadataPath))
+            {
+                // parse metadata to get actual SNAPSHOT version
+                String actualVersion = getMetadataVersion(localRepoMetadataPath, coords);
+                super.download(coords.toCentralURI(actualVersion), destination);
+            }
+        }
+        else
+        {
+            super.download(coords.toCentralURI(), destination);
+        }
+    }
+
+    private boolean isMetadataStale(Path localRepoMetadataPath)
+    {
+        if (!Files.exists(localRepoMetadataPath))
+        {
+            // doesn't exist, it's stale.
+            return true;
+        }
+
+        try
+        {
+            MavenMetadata mavenMetadata = new MavenMetadata(localRepoMetadataPath);
+            return MavenMetadata.isExpiredTimestamp(mavenMetadata.getLastUpdated());
```

Comment:
maybe we could have some configuration à la `mvn -U` to force downloading SNAPSHOT as here we have a sort of hardcoded value (the day before) maybe a sys property?

Reply:
Forcing download of new snapshots could be useful, yes.
Lets make that a new PR after this one gets merged.

Reply:
This ability, to support SNAPSHOT artifacts, is currently only useful to us when testing things.
The actual 10.0.0 release wouldn't have this requirement.
So I think it's safe to push this `--update-snapshots` feature to a future Jetty 10.0.x release.